openwisp_controller/config/base/config.py

```diff
@@ -185,6 +186,45 @@ def _get_templates_from_pk_set(cls, pk_set):
             templates = pk_set
         return templates
 
+    @classmethod
+    def validate_duplicate_vpn_templates(cls, action, instance, templates):
+        """
+        Validates if there are duplicate templates for the same VPN server.
+        Raises a ValidationError if duplicates are found.
+        """
+        if action != 'pre_add':
```

Comment:
This check makes the following test fail - when users switches the template of the device and both the templates belongs to the same VPN. 

It happens because `pre_add` action is sent before `pre_clear`, thus the db still has old template assigned to the config and this logic raises an error.


https://github.com/openwisp/openwisp-controller/blob/ccf380c44ba510a151a7e6f0a20b7742e32c9bb1/openwisp_controller/config/tests/test_admin.py#L2056-L2163